.github/workflows/main.yml

```diff
@@ -7,25 +7,36 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
-      with:
-        java-version: 1.8
-    - uses: derongan/nmsaction@v1
-      with:
-        rev: 1.15
-    - name: Install nms into m2
-      run: mkdir -p $HOME/.m2/repository && cp -a nms-build/.m2/repository/. $HOME/.m2/repository
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-    - name: Build with Gradle
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: ./gradlew shadowJar
-    - name: Upload artifact
-      uses: actions/upload-artifact@v1.0.0
-      with:
-        name: package
-        # Directory containing files to upload
-        path: build/libs
\ No newline at end of file
+      - uses: actions/checkout@v2
+      - name: Load gradle cache
+        uses: actions/cache@v1
+        id: cache
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Install NMS
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
Not sure if we specifically need to specify the action name to be able to use this if statement, but I haven't seen examples like that. Would it be possible to run this check from within the action itself? Perhaps it's not even worth doing that way, since it'd force people to use a specific id for their cache action.

Reply:
this seems fine but Im a bit skeptical about copying the gradle cache rather than the m2 from NMS.... We use snapshot on some deps (spigot) that we would want to refresh